src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -612,17 +612,25 @@
     (when-not (empty? hostnames)
       (map str/trim (str/split hostnames #",")))))
 
-(schema/defn create-dns-alt-names-ext :- Extension
-  "Given a hostname and a comma-separated list of DNS names, create a Subject
-   Alternative Names extension. If there are no alt names provided then
-   defaults will be used."
+(schema/defn create-subject-alt-names-ext :- Extension
+  "Given a hostname and a comma-separated list of DNS (and possibly IP) alt names,
+   create a Subject Alternative Names extension. If there are no alt names
+   provided then defaults will be used."
   [host-name :- schema/Str
    alt-names :- schema/Str]
-  (let [alt-names-list    (split-hostnames alt-names)
-        default-alt-names ["puppet"]]
-    (if-not (empty? alt-names-list)
-      (utils/subject-dns-alt-names (conj alt-names-list host-name) false)
-      (utils/subject-dns-alt-names (conj default-alt-names host-name) false))))
+  (let [split-alt-names (split-hostnames alt-names)
+        default-alt-names ["puppet"]
+        alt-names-list (reduce (fn [acc alt-name]
+                                 (if (str/starts-with? alt-name "IP:")
+                                   (update acc :ip conj (str/replace alt-name #"^IP:" ""))
+                                   (update acc :dns-name conj (str/replace alt-name #"^DNS:" ""))))
+                               {:ip [] :dns-name []} split-alt-names)
+        min-alt-names-list (if (empty? (get alt-names-list :ip))
+                             (dissoc alt-names-list :ip)
```

Comment:
What happens if we don't `dissoc` here?

Reply:
Just tested and I don't need to do that (if it comes in as an empty vector, its just ignores it :)